fog/view/enclave/impl/src/oblivious_utils.rs

```diff
@@ -56,23 +57,14 @@ fn maybe_overwrite_tx_out_search_result(
         shard_tx_out_search_result,
     );
     let shard_ciphertext_length = shard_tx_out_search_result.ciphertext.len();
-
-    let shard_cipher_text_length_delta =
-        u8::try_from(CLIENT_CIPHERTEXT_LENGTH - shard_ciphertext_length)
-            .expect("Shard ciphertext length exceeds bounds");
-    // Need to add a 1 because the first byte is reserved for the delta.
-    assert!(
-        shard_cipher_text_length_delta >= 1,
-        "Shard ciphertext has unexpected length"
-    );
-    client_tx_out_search_result.ciphertext[0].conditional_assign(
-        &shard_cipher_text_length_delta,
-        should_overwrite_tx_out_search_result,
-    );
+    client_tx_out_search_result
+        .payload_length
+        .conditional_assign(
+            &(shard_ciphertext_length as u32),
+            should_overwrite_tx_out_search_result,
+        );
     for idx in 0..shard_ciphertext_length {
-        // Offset the client ciphertext by 1 because the first byte is reserved for the
-        // length delta.
-        client_tx_out_search_result.ciphertext[idx + 1].conditional_assign(
+        client_tx_out_search_result.ciphertext[idx].conditional_assign(
```

Comment:
Is there a conditional assign for slices that we could use instead of doing this byte-at-a-time here?

Reply:
No, I don't think there is. That's why we decided to do it this way.